File: speed/solve.rb
require 'mybench'

n = 500
m = NArray.float(n,n).indgen!
m = m % (n+1) + 1
m = NMatrix.ref(m).transpose
x = NMatrix.float(n,n).indgen!(1).transpose
#x = NVector.float(n).indgen!(1)
y = 0

puts 'm ='
p m
puts 'x ='
p x

printf "solving  y=x/m ...\n",n,n
bench_time(1){ y = x/m }
#  Time: 5.87 sec

puts 'y='
p y

File: speed/solve.py
from mybench import *
from LinearAlgebra import *

n = 500

a = reshape(arrayrange(n*n)%(n+1)+1,(n,n)).astype(Float64)

b = reshape(arrayrange(n*n)+1,(n,n)).astype(Float64)
#b = arrayrange(n).astype(Float64)+1

print "LU factorize & solve %ix%i matrix ..."%(n,n)

def bench_body(a=a,b=b): c=solve_linear_equations(a,b)
bench_time(bench_body,1)

# Time:   8.120 sec
